soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -515,8 +542,16 @@ void CheckTrackerTransition(uint32_t sceneNum) {
 
 void CheckTrackerFrame() {
     if (IS_RANDO) {
-        hideShopRightChecks = CVarGetInteger("gCheckTrackerOptionHideRightShopChecks", 1);
-        alwaysShowGS = CVarGetInteger("gCheckTrackerOptionAlwaysShowGSLocs", 0);
+        bool hideShopRightChecks_ = CVarGetInteger("gCheckTrackerOptionHideRightShopChecks", 1);
+        if (hideShopRightChecks != hideShopRightChecks_) {
+            hideShopRightChecks = hideShopRightChecks_;
+            RecalculateAreaTotals();
+        }
```

Comment:
would changing https://github.com/HarbourMasters/Shipwright/blob/807b00a058be1d9b9f19e1b787e93c6dba5803d1/soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp#L1523 to be

```cpp
if (UIWidgets::EnhancementCheckbox("Hide right-side shop item checks", "gCheckTrackerOptionHideRightShopChecks", false, "", UIWidgets::CheckboxGraphics::Cross, true)) {
    RecalculateAreaTotals();
}
```
no work for this?

Reply:
No, because it's calling the function directly, whereas the helper variables that hold whether Hide Right Side or Show GS are enabled aren't updated until the next frame, so the counts were being inverted.

Reply:
Ohhh, you know what? Maybe changing the helper variable in that if just before calling recalculate might be a better option...

Reply:
> Ohhh, you know what? Maybe changing the helper variable in that if just before calling recalculate might be a better option...

if that works that'd be awesome!

Reply:
Yep, that means the check tracker is down to one per-frame operation, just waiting on a hook for that.